include/nodejs/node_osrm_support.hpp

```diff
@@ -52,30 +53,40 @@ struct PluginParameters
 
 using ObjectOrString = typename mapbox::util::variant<osrm::json::Object, std::string>;
 
-template <typename ResultT> inline v8::Local<v8::Value> render(const ResultT &result);
+template <typename ResultT> inline Napi::Value render(const Napi::Env &env, const ResultT &result);
 
-template <> v8::Local<v8::Value> inline render(const std::string &result)
+template <> Napi::Value inline render(const Napi::Env &env, const std::string &result)
 {
-    return Nan::CopyBuffer(result.data(), result.size()).ToLocalChecked();
+    return Napi::Buffer<char>::Copy(env, result.data(), result.size());
 }
 
-template <> v8::Local<v8::Value> inline render(const ObjectOrString &result)
+template <> Napi::Value inline render(const Napi::Env &env, const ObjectOrString &result)
 {
     if (result.is<osrm::json::Object>())
     {
         // Convert osrm::json object tree into matching v8 object tree
-        v8::Local<v8::Value> value;
-        renderToV8(value, result.get<osrm::json::Object>());
+        Napi::Value value;
+        renderToV8(env, value, result.get<osrm::json::Object>());
         return value;
     }
     else
     {
         // Return the string object as a node Buffer
-        return Nan::CopyBuffer(result.get<std::string>().data(), result.get<std::string>().size())
-            .ToLocalChecked();
+        return Napi::Buffer<char>::Copy(
+            env, result.get<std::string>().data(), result.get<std::string>().size());
     }
 }
 
+inline bool IsUnsignedInteger(const Napi::Value &value)
```

Comment:
Node API doesn't distinguish integers and floats, so I had to add this to be compatible with some input parameter checks we had before.